src/client.js

```diff
@@ -84,9 +92,27 @@ export default class StitchClient {
       authOptions.codec = options.authCodec;
     }
 
-    this.auth = new Auth(this, this.authUrl, authOptions);
-    this.auth.handleRedirect();
-    this.auth.handleCookie();
+    const authFactory = AuthFactory.create(stitchClient, stitchClient.authUrl, authOptions);
+    authFactory.then(auth => {
+      stitchClient.auth = auth;
+      stitchClient.auth.handleRedirect();
```

Comment:
[major] These two return promises.
[minor] Same comment as before about returning a `Promise.all` instead of creating a new Promise at the beginning of this function. Last `.then` can return the stitchClient

Reply:
👍